src/components/magicleap-controls.js

```diff
@@ -0,0 +1,259 @@
+var bind = require('../utils/bind');
+var registerComponent = require('../core/component').registerComponent;
+// var THREE = require('../lib/three');
+
+var trackedControlsUtils = require('../utils/tracked-controls');
+var checkControllerPresentAndSetup = trackedControlsUtils.checkControllerPresentAndSetup;
+var emitIfAxesChanged = trackedControlsUtils.emitIfAxesChanged;
+var onButtonEvent = trackedControlsUtils.onButtonEvent;
+
+var isWebXRAvailable = require('../utils/').device.isWebXRAvailable;
+
+var GAMEPAD_ID_WEBXR = 'magic-leap';
+var GAMEPAD_ID_WEBVR = 'OpenVR ';
+
+var GAMEPAD_ID_PREFIX = isWebXRAvailable ? GAMEPAD_ID_WEBXR : GAMEPAD_ID_WEBVR;
+
+// TODO: Add/Replace with MagicLeap Models
+// var MAGICLEAP_CONTROLLER_MODEL_OBJ_URL = 'https://cdn.aframe.io/controllers/magicleap/controller_ml.obj';
+// var MAGICLEAP_CONTROLLER_MODEL_OBJ_MTL = 'https://cdn.aframe.io/controllers/magicleap/controller_ml.mtl';
+var MAGICLEAP_CONTROLLER_MODEL_OBJ_URL = 'https://cdn.aframe.io/controllers/vive/vr_controller_vive.obj';
+var MAGICLEAP_CONTROLLER_MODEL_OBJ_MTL = 'https://cdn.aframe.io/controllers/vive/vr_controller_vive.mtl';
+
+/**
+ * Button IDs:
+ * 0 - trackpad
+ * 1 - trigger (intensity value from 0.5 to 1)
+ * 2 - grip
+ * 3 - menu (dispatch but better for menu options)
+ *
+ * Axis:
+ * 0 - touchpad x axis
+ * 1 - touchpad y axis
+ */
+var INPUT_MAPPING_WEBVR = {
+  axes: {trackpad: [0, 1]},
+  buttons: ['trigger', 'grip', 'trackpad', 'menu']
+};
+
+/**
+ * Button IDs:
+ * 0 - trigger
+ * 1 - squeeze
+ * 2 - touchpad
+ * 4 - menu (never dispatched on this layer)
+ *
+ * Axis:
+ * 0 - touchpad x axis
+ * 1 - touchpad y axis
+ */
+var INPUT_MAPPING_WEBXR = {
+  axes: {trackpad: [0, 1]},
+  buttons: ['trigger', 'grip', 'trackpad', 'menu']
+};
+
+var INPUT_MAPPING = isWebXRAvailable ? INPUT_MAPPING_WEBXR : INPUT_MAPPING_WEBVR;
```

Comment:
We can omit all WebVR mappings since I don’t think there’s any browser shipping the old API on Magic Leap